File: DSA-loan-management/dsa-mgmt-fe/src/services/eligibility.ts
import apiClient from './apiClient'
import { API_ENDPOINT_NAMES } from '../constants/apiEndpoints'
import { ApiResponse, EligibilityStatus } from '../types'

export interface EligibilityResult {
  applicationId: number
  uniqueCustomerId?: string
  customerName?: string
  email?: string
  mobile?: string
  age?: number
  gender?: string
  location?: string
  employmentType?: string
  productName?: string
  productType?: string
  status: EligibilityStatus
  missingFields?: string[]

  requestedAmount?: number
  eligibleAmount?: number
  proposedEmi?: number
  monthlyIncome?: number
  cibilScore?: number
  interestRatePct?: number
  femaleRebateApplied?: boolean
  tenureYears?: number
  preferredTenure?: number
  foirPct?: number
  ltvPct?: number
  maxAllowedLtvPct?: number

  positiveFactors?: string[]
  reductionNotes?: string[]
  rejections?: string[]

  aiExplanation?: string
}

export const fetchEligibility = async (applicationId: number): Promise<EligibilityResult> => {
  const res = await apiClient.get<ApiResponse<EligibilityResult>>(API_ENDPOINT_NAMES.ELIGIBILITY.EVALUATE, {
    params: { applicationId },
  })
  return res.data?.result ?? res.data
}

File: DSA-loan-management/dsa-mgmt-be/app/schemas/eligibility.py
from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any


class EligibilityEvaluateRequest(BaseModel):
    applicationId: int = Field(..., description="ID of the loan application to evaluate")


class ApplicantSnapshot(BaseModel):
    name: Optional[str] = None
    email: Optional[str] = None
    mobile: Optional[str] = None
    age: Optional[int] = None
    gender: Optional[str] = None
    location: Optional[str] = None
    employmentType: Optional[str] = None
    monthlyIncome: Optional[float] = 0.0
    cibilScore: Optional[int] = None
    loanAmountRequired: Optional[float] = 0.0
    preferredTenure: Optional[int] = None


class EligibilityResponse(BaseModel):
    applicationId: int
    uniqueCustomerId: Optional[str] = None
    customerName: Optional[str] = None
    email: Optional[str] = None
    mobile: Optional[str] = None
    age: Optional[int] = None
    gender: Optional[str] = None
    location: Optional[str] = None
    employmentType: Optional[str] = None
    productName: Optional[str] = None
    productType: Optional[str] = None
    status: str  # ELIGIBLE | PARTIALLY_ELIGIBLE | NOT_ELIGIBLE | INCOMPLETE_DETAILS

    # Financial & Underwriting metrics consumed by UI
    requestedAmount: Optional[float] = 0.0
    eligibleAmount: Optional[float] = 0.0
    proposedEmi: Optional[float] = 0.0
    monthlyIncome: Optional[float] = 0.0
    cibilScore: Optional[int] = None
    interestRatePct: Optional[float] = 0.0
    femaleRebateApplied: Optional[bool] = False
    tenureYears: Optional[int] = 0
    preferredTenure: Optional[int] = None
    foirPct: Optional[float] = 0.0
    ltvPct: Optional[float] = None
    maxAllowedLtvPct: Optional[float] = None

    # Audit & Underwriting lists
    positiveFactors: List[str] = []
    reductionNotes: List[str] = []
    rejections: List[str] = []
    missingFields: List[str] = []

    # AI Summary
    aiExplanation: Optional[str] = None
